tests/e2e/common/common.go

```diff
@@ -665,14 +668,24 @@ func waitAllPodsRunning(t *testing.T, namespace string, done, podsRunning chan s
 			Limit: 100,
 		})
 		require.NoError(t, err, "error getting pods list from k8s")
-		count := 0
+		countOfReadyPods := 0
 		for _, item := range list.Items {
 			// Check pods running, and containers ready.
-			if item.Status.Phase == core_v1.PodRunning && len(item.Status.ContainerStatuses) != 0 && item.Status.ContainerStatuses[0].Ready {
-				count++
+			if item.Status.Phase == core_v1.PodRunning && len(item.Status.ContainerStatuses) != 0 {
+				size := len(item.Status.ContainerStatuses)
+				for _, status := range item.Status.ContainerStatuses {
+					if status.Ready {
+						size--
+					}
+				}
+				if size == 0 {
+					countOfReadyPods++
+				}
 			}
 		}
-		if len(list.Items) == count {
+		if haEnabled && len(list.Items) == countOfReadyPods && countOfReadyPods == numberOfExpectedPodsHAMode {
+			podsRunning <- struct{}{}
+		} else if !haEnabled && len(list.Items) == countOfReadyPods && countOfReadyPods == numberOfExpectedPodsNonHAMode {
```

Comment:
```suggestion
		if len(list.Items) == countOfReadyPods 
		&& (
		  ( haEnabled && countOfReadyPods == numberOfExpectedPodsHAMode ) 
		  || ( !haEnabled && countOfReadyPods == numberOfExpectedPodsNonHAMode )
		) {
			podsRunning <- struct{}{}
		} 
```

Reply:
can change `numberOfExpectedPodsHAMode` to `numHAPods` and 
`numberOfExpectedPodsNonHAMode` to `numNonHAPods`